socorro/app/fetch_transform_save_app.py

```diff
@@ -104,32 +110,140 @@ def get_application_defaults():
 
         return {
             'source.crashstorage_class':
-                'socorro.external.fs.crashstorage.FSPermanentStorage',
+            'socorro.external.fs.crashstorage.FSPermanentStorage',
             'destination.crashstorage_class':
-                'socorro.external.fs.crashstorage.FSPermanentStorage',
+            'socorro.external.fs.crashstorage.FSPermanentStorage',
         }
 
-
     #--------------------------------------------------------------------------
     def __init__(self, config):
         super(FetchTransformSaveApp, self).__init__(config)
         self.waiting_func = None
+        # select the iterator type based on the "number_of_submissions" config
+        if config.number_of_submissions == 'forever':
+            self.source_iterator = self._infinite_iterator
+        elif config.number_of_submissions == 'all':
+            self.source_iterator = self._all_iterator
+        else:
+            self.source_iterator = self._limited_iterator
```

Comment:
You could have just...:
```python
self.source_iterator = {
    'forever': self._infinite_iterator,
    'all': self._all_iterator,
}.get(config.number_of_submissions, self._limited_iterator)
```
An advantage is that you don't have to repeat `self.source_iterator` 3 times.